vault/identity_store_oidc_test.go

```diff
@@ -211,21 +230,19 @@ func TestOIDC_Path_OIDCKeyKey(t *testing.T) {
 	})
 	expectSuccess(t, resp, err)
 
-	// Read "test-key" again
-	respReadTestKeyAfterUpdate, err2 := c.identityStore.HandleRequest(ctx, &logical.Request{
+	// Read "test-key" again and validate
+	resp, err = c.identityStore.HandleRequest(ctx, &logical.Request{
 		Path:      "oidc/key/test-key",
 		Operation: logical.ReadOperation,
 		Storage:   storage,
 	})
-	expectSuccess(t, respReadTestKeyAfterUpdate, err2)
-
-	// Compare response for "test-key" before and after it was updated
-	expectedDiff := map[string]interface{}{
-		"Data.map[rotation_period]: 86400 != 600":   true, // from 24h to 10m
-		"Data.map[verification_ttl]: 86400 != 3600": true, // from 24h to 1h
+	expectSuccess(t, resp, err)
+	if resp.Data["rotation_period"].(int64) != 600 {
```

Comment:
I had no problem with using `deep` (we use it many place), just this particular application of it, which was testing the full diff contents. Typical use would be like:
```
expected := map[string]interface{}{
	"rotation_period": 600,
	"verification_ttl": 3600,
}

if diff := deep.Diff(expected, resp.Data); diff != nil {
	t.Fatal(diff)
}  
```